controllers/apps/components/replication/replication.go

```diff
@@ -180,86 +175,93 @@ func (r *ReplicationSet) HandleRestart(ctx context.Context, obj client.Object) (
 	return vertexes, nil
 }
 
+// HandleRoleChange is the implementation of the type Component interface method, which is used to handle the role change of the Replication workload.
 func (r *ReplicationSet) HandleRoleChange(ctx context.Context, obj client.Object) ([]graph.Vertex, error) {
-	podList, err := r.getRunningPods(ctx, obj)
+	podList, err := util.GetRunningPods(ctx, r.Cli, obj)
 	if err != nil {
 		return nil, err
 	}
 	if len(podList) == 0 {
 		return nil, nil
 	}
-	vertexes := make([]graph.Vertex, 0)
-	podsToSyncStatus := make([]*corev1.Pod, 0)
-	for i := range podList {
-		pod := &podList[i]
-		// if there is no role label on the Pod, it needs to be updated with statefulSet's role label.
-		if v, ok := pod.Labels[constant.RoleLabelKey]; !ok || v == "" {
-			_, o := util.ParseParentNameAndOrdinal(pod.Name)
-			role := string(Secondary)
-			if o == r.getPrimaryIndex() {
-				role = string(Primary)
-			}
-			pod.GetLabels()[constant.RoleLabelKey] = role
-			vertexes = append(vertexes, &ictrltypes.LifecycleVertex{
-				Obj:    pod,
-				Action: ictrltypes.ActionUpdatePtr(), // update or patch?
-			})
-		}
-		// else {
-		//	podsToSyncStatus = append(podsToSyncStatus, pod)
-		// }
-		podsToSyncStatus = append(podsToSyncStatus, pod)
-	}
-	// // REVIEW/TODO: (Y-Rookie)
-	// //  1. should employ rolling deletion as default strategy instead of delete them all.
-	// if err := util.DeleteStsPods(ctx, r.Cli, sts); err != nil {
-	// 	return err
-	// }
-	// sync cluster.spec.componentSpecs.[x].primaryIndex when failover occurs and switchPolicy is Noop.
-	// TODO(refactor): syncPrimaryIndex will update cluster spec, resolve it.
-	if err := syncPrimaryIndex(ctx, r.Cli, r.Cluster, r.getName()); err != nil {
+	// sync pod role label and annotations
+	vertexes, err := r.asyncReplicationPodRoleLabelAndAnnotations(podList)
+	if err != nil {
 		return nil, err
 	}
-	// sync cluster.status.components.replicationSet.status
-	if err := syncReplicationSetClusterStatus(r.Cluster, r.getWorkloadType(), r.getName(), podsToSyncStatus); err != nil {
+	// async cluster.status.components.replicationSet.status
+	if err := asyncReplicationSetClusterStatus(r.Cluster, r.getWorkloadType(), r.getName(), podList); err != nil {
```

Comment:
the podList used here is obtained from k8s directly, not the one updated above, it means the newly resolved role labels will not take effect in current reconciliation

Reply:
> the podList used here is obtained from k8s directly, not the one updated above, it means the newly resolved role labels will not take effect in current reconciliation

I think this is acceptable. The update of the pod role above only occurs when the pod is just created and has no label.